cms/static/js/views/utils/view_utils.js

```diff
@@ -173,6 +173,34 @@ define(["jquery", "underscore", "gettext", "js/views/feedback_notification", "js
             return false;
         };
 
+        keywordValidator = (function () {
+            var regexp = /%%+[^%]+%%/g;
+            var keywords = ['%%USER_ID%%', '%%USER_FULLNAME%%', '%%COURSE_DISPLAY_NAME%%', '%%COURSE_END_DATE%%'];
+            var validate = function (string) {
+                var regex_match = string.match(regexp);
+                var found_keywords = regex_match == null ? [] : regex_match;
+                var invalid_keywords = [];
+                var num_found = found_keywords.length;
+                var curr_keyword;
+
+                for (var i = 0; i < num_found; i++) {
```

Comment:
You're probably better off using [`$.each`](http://api.jquery.com/jQuery.each/) to iterate over `found_keywords`, rather than using a `for` loop -- `$.each` makes it clearer what you're trying to do.